src/components/visualizations/chart/test/loadEvent.spec.ts

```diff
@@ -0,0 +1,84 @@
+// (C) 2007-2018 GoodData Corporation
+import handleChartLoad from '../events/load';
+
+const buildChart = (
+    axisGroupAddClass: () => {},
+    labelGroupAddClass: () => {},
+    opposite: boolean
+) => ({
+    yAxis: [
+        {
+            axisGroup: {
+                addClass: axisGroupAddClass
+            },
+            labelGroup: {
+                addClass: labelGroupAddClass
+            },
+            opposite
+        }
+    ]
+});
+
+const addSecondaryAxisToChart = (
+    chart: any
+) => {
+    const { yAxis: [ primaryAxis ] } = chart;
+    return {
+        yAxis: [
+            primaryAxis,
+            {
+                ...primaryAxis,
+                opposite: !primaryAxis.opposite
+            }
+        ]
+    };
+};
+
+const executeTest = (opposite: boolean, isDualAxesChart: boolean) => {
+    const axisGroupAddClass = jest.fn();
+    const labelGroupAddClass = jest.fn();
+
+    let chart = buildChart(axisGroupAddClass, labelGroupAddClass, opposite);
```

Comment:
Please pass `isDualAxisChart` argument down to `buildChart()` instead of `opposite` argument. There is no need to have additional function `addSecondaryAxisToChart()`.

Reply:
I mean something like:
```
const buildChart = (
    axisGroupAddClass: () => {},
    labelGroupAddClass: () => {},
    isDualAxisChart: boolean
) => {
    const getAxis = (opposite: boolean) => ({
        {
            axisGroup: {
                addClass: axisGroupAddClass
            },
            labelGroup: {
                addClass: labelGroupAddClass
            },
            opposite
        }
    });

    const yAxis = [
        getAxis(false)
    ];

    if (isDualAxisChart) {
        yAxis.push(getAxis(true))
    }

    return { yAxis };
};

let chart = buildChart(axisGroupAddClass, labelGroupAddClass, isDualAxisChart);
```